content/200-orm/200-prisma-client/400-type-safety/100-operating-against-partial-structures-of-model-types.mdx

```diff
@@ -128,10 +128,22 @@ const usersWithPosts: UsersWithPosts = await getUsersWithPosts()
 
 ### Solution
 
-With the `PromiseReturnType` that is exposed by the `Prisma` namespace, you can solve this more elegantly:
+You can use native the TypeScript utility type [`Awaited`](https://www.typescriptlang.org/docs/handbook/utility-types.html#awaitedtype) and [`ReturnType`](https://www.typescriptlang.org/docs/handbook/utility-types.html#returntypetype) to solve the problem elegantly:
+
+```ts
+type UsersWithPosts = Prisma.Awaited<ReturnType<typeof getUsersWithPosts>>
```

Comment:
This should be `Awaited` and not `Prisma.Awaited`, it's a built-in type in TypeScript.